pkg/quota/evaluator/core/pods.go

```diff
@@ -115,10 +121,32 @@ func (p *podEvaluator) Constraints(required []api.ResourceName, item runtime.Obj
 	requiredSet := quota.ToSet(required)
 	missingSet := sets.NewString()
 	for i := range pod.Spec.Containers {
-		enforcePodContainerConstraints(&pod.Spec.Containers[i], requiredSet, missingSet)
+		if len(pod.Spec.Containers[i].VolumeMounts) == 0 {
```

Comment:
This looks a little verbose and duplicate for both containers and init container check; why can't you just have `enforcePodContainerConstraints` takes care of checking volume mounts since you already passed container[i] into it?